app/interactors/new_document/document_type_selection_interactor.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+class NewDocument::DocumentTypeSelectionInteractor < ApplicationInteractor
+  delegate :params,
+           :user,
+           :redirect_url,
+           :document,
+           :needs_refining,
+           :current_selection,
+           :previous_selection,
+           to: :context
+
+
+  def call
+    find_previous_selection
```

Comment:
It would be more correct to rename `current_selection` to `selection`. "page" and "next_page" would probably make more sense as names, but felt wrong.